The runtime unit test failure looks legit:
```
 	 # github.com/cilium/cilium/pkg/fqdn/dnsproxy [github.com/cilium/cilium/pkg/fqdn/dnsproxy.test]
	 pkg/fqdn/dnsproxy/proxy_test.go:186:3: cannot use func literal (type func(net.IP) (ipcache.Identity, bool, error)) as type LookupSecIDByIPFunc in argument to StartDNSProxy
```

--- a/daemon/cmd/daemon.go
+++ b/daemon/cmd/daemon.go
@@ -23,7 +23,6 @@ import (
 	"sync"
 	"time"
 
-	hubbleProto "github.com/cilium/cilium/api/v1/flow"
 	"github.com/cilium/cilium/api/v1/models"
 	health "github.com/cilium/cilium/cilium-health/launch"
 	"github.com/cilium/cilium/pkg/bpf"
@@ -40,18 +39,15 @@ import (
 	"github.com/cilium/cilium/pkg/endpointmanager"
 	"github.com/cilium/cilium/pkg/envoy"
 	"github.com/cilium/cilium/pkg/fqdn"
-	hubbleV1 "github.com/cilium/cilium/pkg/hubble/api/v1"
 	"github.com/cilium/cilium/pkg/hubble/observer"
 	"github.com/cilium/cilium/pkg/identity"
 	"github.com/cilium/cilium/pkg/identity/cache"
 	"github.com/cilium/cilium/pkg/identity/identitymanager"
-	identitymodel "github.com/cilium/cilium/pkg/identity/model"
 	"github.com/cilium/cilium/pkg/ipam"
 	ipamOption "github.com/cilium/cilium/pkg/ipam/option"
 	"github.com/cilium/cilium/pkg/ipcache"
 	"github.com/cilium/cilium/pkg/k8s"
 	"github.com/cilium/cilium/pkg/k8s/watchers"
-	"github.com/cilium/cilium/pkg/loadbalancer"
 	"github.com/cilium/cilium/pkg/lock"
 	"github.com/cilium/cilium/pkg/logging"
 	"github.com/cilium/cilium/pkg/logging/logfields"
@@ -656,61 +652,4 @@ func (d *Daemon) GetNodeSuffix() string {
 	}
 
 	return ip.String()
-}
-
-// GetIdentity looks up identity by ID from Cilium's identity cache. Hubble uses the identity info
-// to populate source and destination labels of flows.
-//
-//  - IdentityGetter: https://github.com/cilium/hubble/blob/04ab72591faca62a305ce0715108876167182e04/pkg/parser/getters/getters.go#L40
-func (d *Daemon) GetIdentity(securityIdentity uint32) (*models.Identity, error) {
-	ident := d.identityAllocator.LookupIdentityByID(context.Background(), identity.NumericIdentity(securityIdentity))
-	if ident == nil {
-		return nil, fmt.Errorf("identity %d not found", securityIdentity)
-	}
-	return identitymodel.CreateModel(ident), nil
-}
-
-// GetEndpointInfo returns endpoint info for a given IP address. Hubble uses this function to populate
-// fields like namespace and pod name for local endpoints.
-//
-//  - EndpointGetter: https://github.com/cilium/hubble/blob/04ab72591faca62a305ce0715108876167182e04/pkg/parser/getters/getters.go#L34
-func (d *Daemon) GetEndpointInfo(ip net.IP) (endpoint hubbleV1.EndpointInfo, ok bool) {
-	ep := d.endpointManager.LookupIP(ip)
-	if ep == nil {
-		return nil, false
-	}
-	return ep, true
-}
-
-// GetNamesOf implements DNSGetter.GetNamesOf. It looks up DNS names of a given IP from the
-// FQDN cache of an endpoint specified by sourceEpID.
-//
-//  - DNSGetter: https://github.com/cilium/hubble/blob/04ab72591faca62a305ce0715108876167182e04/pkg/parser/getters/getters.go#L27
-func (d *Daemon) GetNamesOf(sourceEpID uint32, ip net.IP) []string {
-	ep := d.endpointManager.LookupCiliumID(uint16(sourceEpID))
-	if ep == nil {
-		return nil
-	}
-	return ep.DNSHistory.LookupIP(ip)
-}
-
-// GetServiceByAddr looks up service by IP/port. Hubble uses this function to annotate flows
-// with service information.
-//
-//  - ServiceGetter: https://github.com/cilium/hubble/blob/04ab72591faca62a305ce0715108876167182e04/pkg/parser/getters/getters.go#L52
-func (d *Daemon) GetServiceByAddr(ip net.IP, port uint16) (hubbleProto.Service, bool) {
-	addr := loadbalancer.L3n4Addr{
-		IP: ip,
-		L4Addr: loadbalancer.L4Addr{
-			Port: port,
-		},
-	}
-	namespace, name, ok := d.svc.GetServiceNameByAddr(addr)
-	if !ok {
-		return hubbleProto.Service{}, false
-	}
-	return hubbleProto.Service{
-		Namespace: namespace,
-		Name:      name,
-	}, true
 }

--- a/daemon/cmd/fqdn.go
+++ b/daemon/cmd/fqdn.go
@@ -394,41 +394,6 @@ func (d *Daemon) lookupEPByIP(endpointIP net.IP) (endpoint *endpoint.Endpoint, e
 	return e, nil
 }
 
-// LookupSecIDByIP returns the security ID for the given IP. If the security ID
-// cannot be found, ok is false.
-func (d *Daemon) LookupSecIDByIP(ip net.IP) (id ipcache.Identity, ok bool) {
-	if ip == nil {
-		return ipcache.Identity{}, false
-	}
-
-	if id, ok = ipcache.IPIdentityCache.LookupByIP(ip.String()); ok {
-		return id, ok
-	}
-
-	ipv6Prefixes, ipv4Prefixes := d.GetCIDRPrefixLengths()
-	prefixes := ipv4Prefixes
-	bits := net.IPv4len * 8
-	if ip.To4() == nil {
-		prefixes = ipv6Prefixes
-		bits = net.IPv6len * 8
-	}
-	for _, prefixLen := range prefixes {
-		if prefixLen == bits {
-			// IP lookup was already done above; skip it here
-			continue
-		}
-		mask := net.CIDRMask(prefixLen, bits)
-		cidr := net.IPNet{
-			IP:   ip.Mask(mask),
-			Mask: mask,
-		}
-		if id, ok = ipcache.IPIdentityCache.LookupByPrefix(cidr.String()); ok {
-			return id, ok
-		}
-	}
-	return id, false
-}
-
 // NotifyOnDNSMsg handles DNS data in the daemon by emitting monitor
 // events, proxy metrics and storing DNS data in the DNS cache. This may
 // result in rule generation.

--- a/daemon/cmd/fqdn.go
+++ b/daemon/cmd/fqdn.go
@@ -301,7 +301,7 @@ func (d *Daemon) bootstrapFQDN(restoredEndpoints *endpointRestoreState, preCache
 	if err != nil {
 		return err
 	}
-	proxy.DefaultDNSProxy, err = dnsproxy.StartDNSProxy("", port, option.Config.ToFQDNsEnableDNSCompression, d.lookupEPByIP, d.lookupSecIDByIP, d.notifyOnDNSMsg)
+	proxy.DefaultDNSProxy, err = dnsproxy.StartDNSProxy("", port, option.Config.ToFQDNsEnableDNSCompression, d.lookupEPByIP, d.LookupSecIDByIP, d.notifyOnDNSMsg)
 	if err == nil {
 		// Increase the ProxyPort reference count so that it will never get released.
 		err = d.l7Proxy.SetProxyPort(listenerName, proxy.DefaultDNSProxy.BindPort)
@@ -394,22 +394,39 @@ func (d *Daemon) lookupEPByIP(endpointIP net.IP) (endpoint *endpoint.Endpoint, e
 	return e, nil
 }
 
-func (d *Daemon) lookupSecIDByIP(ip net.IP) (secID ipcache.Identity, exists bool, err error) {
-	ipv6Prefixes, ipv4Prefixes := d.prefixLengths.ToBPFData()
-	prefixes := ipv4Prefixes
-	if ip.To4() == nil {
-		prefixes = ipv6Prefixes
+// LookupSecIDByIP returns the security ID for the given IP. If the security ID
+// cannot be found, ok is false.
+func (d *Daemon) LookupSecIDByIP(ip net.IP) (id ipcache.Identity, ok bool) {
+	if ip == nil {
+		return ipcache.Identity{}, false
 	}
 
+	if id, ok = ipcache.IPIdentityCache.LookupByIP(ip.String()); ok {
+		return id, ok
+	}
+
+	ipv6Prefixes, ipv4Prefixes := d.GetCIDRPrefixLengths()
+	prefixes := ipv4Prefixes
+	bits := net.IPv4len * 8
+	if ip.To4() == nil {
+		prefixes = ipv6Prefixes
+		bits = net.IPv6len * 8
+	}
 	for _, prefixLen := range prefixes {
-		maskedStr := fmt.Sprintf("%s/%d", ip, prefixLen)
-		_, cidr, _ := net.ParseCIDR(maskedStr)
-		secID, exists = ipcache.IPIdentityCache.LookupByPrefix(cidr.String())
-		if exists == true {
-			break
+		if prefixLen == bits {
+			// IP lookup was already done above; skip it here
+			continue
+		}
+		mask := net.CIDRMask(prefixLen, bits)
+		cidr := net.IPNet{
+			IP:   ip.Mask(mask),
+			Mask: mask,
+		}
+		if id, ok = ipcache.IPIdentityCache.LookupByPrefix(cidr.String()); ok {
+			return id, ok
 		}
 	}
-	return secID, exists, nil
+	return id, false
 }
 
 // NotifyOnDNSMsg handles DNS data in the daemon by emitting monitor

--- a/daemon/cmd/hubble.go
+++ b/daemon/cmd/hubble.go
@@ -16,10 +16,14 @@ package cmd
 
 import (
 	"context"
+	"fmt"
+	"net"
 	"time"
 
+	flowpb "github.com/cilium/cilium/api/v1/flow"
 	"github.com/cilium/cilium/api/v1/models"
 	observerpb "github.com/cilium/cilium/api/v1/observer"
+	v1 "github.com/cilium/cilium/pkg/hubble/api/v1"
 	"github.com/cilium/cilium/pkg/hubble/listener"
 	"github.com/cilium/cilium/pkg/hubble/metrics"
 	"github.com/cilium/cilium/pkg/hubble/observer"
@@ -28,7 +32,10 @@ import (
 	"github.com/cilium/cilium/pkg/hubble/peer"
 	"github.com/cilium/cilium/pkg/hubble/server"
 	"github.com/cilium/cilium/pkg/hubble/server/serveroption"
+	"github.com/cilium/cilium/pkg/identity"
+	identitymodel "github.com/cilium/cilium/pkg/identity/model"
 	"github.com/cilium/cilium/pkg/ipcache"
+	"github.com/cilium/cilium/pkg/loadbalancer"
 	"github.com/cilium/cilium/pkg/logging"
 	"github.com/cilium/cilium/pkg/logging/logfields"
 	"github.com/cilium/cilium/pkg/option"
@@ -157,4 +164,61 @@ func (d *Daemon) launchHubble() {
 			return
 		}
 	}
+}
+
+// GetIdentity looks up identity by ID from Cilium's identity cache. Hubble uses the identity info
+// to populate source and destination labels of flows.
+//
+//  - IdentityGetter: https://github.com/cilium/hubble/blob/04ab72591faca62a305ce0715108876167182e04/pkg/parser/getters/getters.go#L40
+func (d *Daemon) GetIdentity(securityIdentity uint32) (*models.Identity, error) {
+	ident := d.identityAllocator.LookupIdentityByID(context.Background(), identity.NumericIdentity(securityIdentity))
+	if ident == nil {
+		return nil, fmt.Errorf("identity %d not found", securityIdentity)
+	}
+	return identitymodel.CreateModel(ident), nil
+}
+
+// GetEndpointInfo returns endpoint info for a given IP address. Hubble uses this function to populate
+// fields like namespace and pod name for local endpoints.
+//
+//  - EndpointGetter: https://github.com/cilium/hubble/blob/04ab72591faca62a305ce0715108876167182e04/pkg/parser/getters/getters.go#L34
+func (d *Daemon) GetEndpointInfo(ip net.IP) (endpoint v1.EndpointInfo, ok bool) {
+	ep := d.endpointManager.LookupIP(ip)
+	if ep == nil {
+		return nil, false
+	}
+	return ep, true
+}
+
+// GetNamesOf implements DNSGetter.GetNamesOf. It looks up DNS names of a given IP from the
+// FQDN cache of an endpoint specified by sourceEpID.
+//
+//  - DNSGetter: https://github.com/cilium/hubble/blob/04ab72591faca62a305ce0715108876167182e04/pkg/parser/getters/getters.go#L27
+func (d *Daemon) GetNamesOf(sourceEpID uint32, ip net.IP) []string {
+	ep := d.endpointManager.LookupCiliumID(uint16(sourceEpID))
+	if ep == nil {
+		return nil
+	}
+	return ep.DNSHistory.LookupIP(ip)
+}
+
+// GetServiceByAddr looks up service by IP/port. Hubble uses this function to annotate flows
+// with service information.
+//
+//  - ServiceGetter: https://github.com/cilium/hubble/blob/04ab72591faca62a305ce0715108876167182e04/pkg/parser/getters/getters.go#L52
+func (d *Daemon) GetServiceByAddr(ip net.IP, port uint16) (flowpb.Service, bool) {
+	addr := loadbalancer.L3n4Addr{
+		IP: ip,
+		L4Addr: loadbalancer.L4Addr{
+			Port: port,
+		},
+	}
+	namespace, name, ok := d.svc.GetServiceNameByAddr(addr)
+	if !ok {
+		return flowpb.Service{}, false
+	}
+	return flowpb.Service{
+		Namespace: namespace,
+		Name:      name,
+	}, true
 }

--- a/daemon/cmd/hubble.go
+++ b/daemon/cmd/hubble.go
@@ -91,7 +91,7 @@ func (d *Daemon) launchHubble() {
 		return
 	}
 
-	payloadParser, err := parser.New(logger, d, d, d, ipcache.IPIdentityCache, d)
+	payloadParser, err := parser.New(logger, d, d, d, d, d)
 	if err != nil {
 		logger.WithError(err).Error("Failed to initialize Hubble")
 		return
@@ -221,4 +221,49 @@ func (d *Daemon) GetServiceByAddr(ip net.IP, port uint16) (flowpb.Service, bool)
 		Namespace: namespace,
 		Name:      name,
 	}, true
+}
+
+// GetK8sMetadata returns the Kubernetes metadata for the given IP address.
+// It implements hubble parser's IPGetter.GetK8sMetadata.
+func (d *Daemon) GetK8sMetadata(ip net.IP) *ipcache.K8sMetadata {
+	if ip == nil {
+		return nil
+	}
+	return ipcache.IPIdentityCache.GetK8sMetadata(ip.String())
+}
+
+// LookupSecIDByIP returns the security ID for the given IP. If the security ID
+// cannot be found, ok is false.
+// It implements hubble parser's IPGetter.LookupSecIDByIP.
+func (d *Daemon) LookupSecIDByIP(ip net.IP) (id ipcache.Identity, ok bool) {
+	if ip == nil {
+		return ipcache.Identity{}, false
+	}
+
+	if id, ok = ipcache.IPIdentityCache.LookupByIP(ip.String()); ok {
+		return id, ok
+	}
+
+	ipv6Prefixes, ipv4Prefixes := d.GetCIDRPrefixLengths()
+	prefixes := ipv4Prefixes
+	bits := net.IPv4len * 8
+	if ip.To4() == nil {
+		prefixes = ipv6Prefixes
+		bits = net.IPv6len * 8
+	}
+	for _, prefixLen := range prefixes {
+		if prefixLen == bits {
+			// IP lookup was already done above; skip it here
+			continue
+		}
+		mask := net.CIDRMask(prefixLen, bits)
+		cidr := net.IPNet{
+			IP:   ip.Mask(mask),
+			Mask: mask,
+		}
+		if id, ok = ipcache.IPIdentityCache.LookupByPrefix(cidr.String()); ok {
+			return id, ok
+		}
+	}
+	return id, false
 }

--- a/pkg/fqdn/dnsproxy/proxy.go
+++ b/pkg/fqdn/dnsproxy/proxy.go
@@ -201,7 +201,7 @@ type LookupEndpointIDByIPFunc func(ip net.IP) (endpoint *endpoint.Endpoint, err 
 // LookupSecIDByIPFunc Func wraps logic to lookup an IP's security ID from the
 // ipcache.
 // See DNSProxy.LookupSecIDByIP for usage.
-type LookupSecIDByIPFunc func(ip net.IP) (secID ipcache.Identity, exists bool, err error)
+type LookupSecIDByIPFunc func(ip net.IP) (secID ipcache.Identity, exists bool)
 
 // NotifyOnDNSMsgFunc handles propagating DNS response data
 // See DNSProxy.LookupEndpointIDByIP for usage.
@@ -396,10 +396,10 @@ func (p *DNSProxy) ServeDNS(w dns.ResponseWriter, request *dns.Msg) {
 		return
 	}
 
-	serverSecID, exists, err := p.LookupSecIDByIP(targetServerIP)
-	if !exists || err != nil {
-		scopedLog.WithError(err).WithField("server", targetServerAddr).Debug("cannot find server ip in ipcache")
-		stat.Err = fmt.Errorf("Cannot find server ip in ipcache: %s", err)
+	serverSecID, exists := p.LookupSecIDByIP(targetServerIP)
+	if !exists {
+		scopedLog.WithField("server", targetServerAddr).Debug("cannot find server ip in ipcache")
+		stat.Err = fmt.Errorf("cannot find security identity for server ip=%s", targetServerIP)
 		stat.ProcessingTime.End(false)
 		p.NotifyOnDNSMsg(time.Now(), ep, epIPPort, targetServerAddr, request, protocol, false, stat)
 		p.sendRefused(scopedLog, w, request)

--- a/pkg/fqdn/dnsproxy/proxy_test.go
+++ b/pkg/fqdn/dnsproxy/proxy_test.go
@@ -183,19 +183,19 @@ func (s *DNSProxyTestSuite) SetUpSuite(c *C) {
 			return endpoint.NewEndpointWithState(s, &endpoint.FakeEndpointProxy{}, &allocator.FakeIdentityAllocator{}, uint16(epID1), endpoint.StateReady), nil
 		},
 		// LookupSecIDByIP
-		func(ip net.IP) (ipcache.Identity, bool, error) {
+		func(ip net.IP) (ipcache.Identity, bool) {
 			DNSServerListenerAddr := (s.dnsServer.Listener.Addr()).(*net.TCPAddr)
 			switch {
 			case ip.String() == DNSServerListenerAddr.IP.String():
 				ident := ipcache.Identity{
 					ID:     dstID1,
 					Source: source.Unspec}
-				return ident, true, nil
+				return ident, true
 			default:
 				ident := ipcache.Identity{
 					ID:     dstID2,
 					Source: source.Unspec}
-				return ident, true, nil
+				return ident, true
 			}
 		},
 		// NotifyOnDNSMsg

--- a/pkg/hubble/parser/getters/getters.go
+++ b/pkg/hubble/parser/getters/getters.go
@@ -45,10 +45,10 @@ type IdentityGetter interface {
 // IPGetter fetches per-IP metadata
 type IPGetter interface {
 	// GetK8sMetadata returns Kubernetes metadata for the given IP address.
-	GetK8sMetadata(ip string) *ipcache.K8sMetadata
-	// LookupByIP returns the corresponding security identity that endpoint IP
-	// maps to as well as if the corresponding entry exists.
-	LookupByIP(ip string) (ipcache.Identity, bool)
+	GetK8sMetadata(ip net.IP) *ipcache.K8sMetadata
+	// LookupSecIDByIP returns the corresponding security identity that
+	// endpoint IP maps to as well as if the corresponding entry exists.
+	LookupSecIDByIP(ip net.IP) (ipcache.Identity, bool)
 }
 
 // ServiceGetter fetches service metadata.

--- a/pkg/hubble/parser/seven/parser.go
+++ b/pkg/hubble/parser/seven/parser.go
@@ -121,10 +121,10 @@ func (p *Parser) Decode(payload *pb.Payload, decoded *pb.Flow) error {
 		destinationNames = p.dnsGetter.GetNamesOf(uint32(sourceEndpoint.ID), destinationIP)
 	}
 	if p.ipGetter != nil {
-		if meta := p.ipGetter.GetK8sMetadata(sourceIP.String()); meta != nil {
+		if meta := p.ipGetter.GetK8sMetadata(sourceIP); meta != nil {
 			sourceNamespace, sourcePod = meta.Namespace, meta.PodName
 		}
-		if meta := p.ipGetter.GetK8sMetadata(destinationIP.String()); meta != nil {
+		if meta := p.ipGetter.GetK8sMetadata(destinationIP); meta != nil {
 			destinationNamespace, destinationPod = meta.Namespace, meta.PodName
 		}
 	}

--- a/pkg/hubble/parser/seven/parser_test.go
+++ b/pkg/hubble/parser/seven/parser_test.go
@@ -123,8 +123,8 @@ func TestDecodeL7HTTPRecord(t *testing.T) {
 		},
 	}
 	IPGetter := &testutils.FakeIPGetter{
-		OnGetK8sMetadata: func(ip string) *ipcache.K8sMetadata {
-			if ip == fakeDestinationEndpoint.IPv4 {
+		OnGetK8sMetadata: func(ip net.IP) *ipcache.K8sMetadata {
+			if ip.String() == fakeDestinationEndpoint.IPv4 {
 				return &ipcache.K8sMetadata{
 					Namespace: "default",
 					PodName:   "pod-1234",

--- a/pkg/hubble/parser/threefour/parser.go
+++ b/pkg/hubble/parser/threefour/parser.go
@@ -255,10 +255,10 @@ func (p *Parser) resolveEndpoint(ip net.IP, securityIdentity uint32) *pb.Endpoin
 	// for remote endpoints, assemble the information via ip and identity
 	var namespace, podName string
 	if p.ipGetter != nil {
-		if ipIdentity, ok := p.ipGetter.LookupByIP(ip.String()); ok {
+		if ipIdentity, ok := p.ipGetter.LookupSecIDByIP(ip); ok {
 			securityIdentity = uint32(ipIdentity.ID)
 		}
-		if meta := p.ipGetter.GetK8sMetadata(ip.String()); meta != nil {
+		if meta := p.ipGetter.GetK8sMetadata(ip); meta != nil {
 			namespace, podName = meta.Namespace, meta.PodName
 		}
 	}

--- a/pkg/hubble/parser/threefour/parser_test.go
+++ b/pkg/hubble/parser/threefour/parser_test.go
@@ -86,8 +86,8 @@ func TestL34Decode(t *testing.T) {
 		},
 	}
 	ipGetter := &testutils.FakeIPGetter{
-		OnGetK8sMetadata: func(ip string) *ipcache.K8sMetadata {
-			if ip == "192.168.33.11" {
+		OnGetK8sMetadata: func(ip net.IP) *ipcache.K8sMetadata {
+			if ip.String() == "192.168.33.11" {
 				return &ipcache.K8sMetadata{
 					Namespace: "remote",
 					PodName:   "pod-192.168.33.11",
@@ -95,9 +95,9 @@ func TestL34Decode(t *testing.T) {
 			}
 			return nil
 		},
-		OnLookupByIP: func(ip string) (ipcache.Identity, bool) {
+		OnLookupSecIDByIP: func(ip net.IP) (ipcache.Identity, bool) {
 			// pretend IP belongs to a pod on a remote node
-			if ip == "192.168.33.11" {
+			if ip.String() == "192.168.33.11" {
 				return ipcache.Identity{
 					ID:     1234,
 					Source: source.Unspec,

--- a/pkg/hubble/testutils/fake.go
+++ b/pkg/hubble/testutils/fake.go
@@ -103,32 +103,32 @@ var NoopEndpointGetter = FakeEndpointGetter{
 
 // FakeIPGetter is used for unit tests that needs IPGetter.
 type FakeIPGetter struct {
-	OnGetK8sMetadata func(ip string) *ipcache.K8sMetadata
-	OnLookupByIP     func(ip string) (ipcache.Identity, bool)
+	OnGetK8sMetadata  func(ip net.IP) *ipcache.K8sMetadata
+	OnLookupSecIDByIP func(ip net.IP) (ipcache.Identity, bool)
 }
 
 // GetK8sMetadata implements FakeIPGetter.GetK8sMetadata.
-func (f *FakeIPGetter) GetK8sMetadata(ip string) *ipcache.K8sMetadata {
+func (f *FakeIPGetter) GetK8sMetadata(ip net.IP) *ipcache.K8sMetadata {
 	if f.OnGetK8sMetadata != nil {
 		return f.OnGetK8sMetadata(ip)
 	}
 	panic("OnGetK8sMetadata not set")
 }
 
-// LookupByIP implements FakeIPGetter.LookupByIP.
-func (f *FakeIPGetter) LookupByIP(ip string) (ipcache.Identity, bool) {
-	if f.OnLookupByIP != nil {
-		return f.OnLookupByIP(ip)
+// LookupByIP implements FakeIPGetter.LookupSecIDByIP.
+func (f *FakeIPGetter) LookupSecIDByIP(ip net.IP) (ipcache.Identity, bool) {
+	if f.OnLookupSecIDByIP != nil {
+		return f.OnLookupSecIDByIP(ip)
 	}
 	panic("OnLookupByIP not set")
 }
 
 // NoopIPGetter always returns an empty response.
 var NoopIPGetter = FakeIPGetter{
-	OnGetK8sMetadata: func(ip string) *ipcache.K8sMetadata {
+	OnGetK8sMetadata: func(ip net.IP) *ipcache.K8sMetadata {
 		return nil
 	},
-	OnLookupByIP: func(ip string) (ipcache.Identity, bool) {
+	OnLookupSecIDByIP: func(ip net.IP) (ipcache.Identity, bool) {
 		return ipcache.Identity{}, false
 	},
 }

